compose/service.py

```diff
@@ -519,6 +522,59 @@ def pull(self, insecure_registry=False):
             )
 
 
+def get_container_data_volumes(container, volumes_option):
+    """Find the container data volumes that are in `volumes_option`, and return
+    a mapping of volume bindings for those volumes.
+    """
+    volumes = []
+
+    volumes_option = volumes_option or []
+    container_volumes = container.get('Volumes') or {}
+    original_volumes = set(volumes_option + container_volumes.keys())
+
+    for volume in original_volumes:
```

Comment:
Have you tried against Docker master?
You should indeed be able to override an existing volume, and volumeFrom have priority over all.